api/v1alpha1/autoscaler_types.go

```diff
@@ -0,0 +1,120 @@
+/*
+
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"github.com/crossplane/crossplane-runtime/apis/core/v1alpha1"
+	runtimev1alpha1 "github.com/crossplane/crossplane-runtime/apis/core/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// Protocol defines network protocols supported for things like container ports.
+type Protocol string
+
+// TriggerType defines the type of trigger
+type TriggerType string
+
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:categories={oam}
+// Autoscaler is the Schema for the autoscalers API
+type Autoscaler struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   AutoscalerSpec   `json:"spec"`
+	Status AutoscalerStatus `json:"status,omitempty"`
+}
+
+func (as *Autoscaler) SetConditions(c ...v1alpha1.Condition) {
+	as.Status.SetConditions(c...)
+}
+
+func (as *Autoscaler) GetCondition(conditionType v1alpha1.ConditionType) v1alpha1.Condition {
+	return as.Status.GetCondition(conditionType)
+}
+
+func (as *Autoscaler) GetWorkloadReference() v1alpha1.TypedReference {
+	return as.Spec.WorkloadReference
+}
+
+func (as *Autoscaler) SetWorkloadReference(reference v1alpha1.TypedReference) {
+	as.Spec.WorkloadReference = reference
+}
+
+// Trigger defines the trigger of Autoscaler
+type Trigger struct {
+	// Name is the trigger name, if not set, it will be automatically generated and make it globally unique
+	Name string `json:"name,omitempty"`
+
+	// Enabled marks whether the trigger immediately. Defaults to `true`
+	Enabled bool `json:"enabled,omitempty"`
```

Comment:
Why have this field?
If it is not enabled, why not just remote it from trigger list?

Reply:
removed